src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/Registration/GssfController.php

```diff
@@ -105,8 +113,8 @@ public function consumeAssertionAction(Request $httpRequest, $provider)
                 sprintf('Could not process received Response, error: "%s"', $exception->getMessage())
             );
 
-            return $this->renderInitiateForm(
-                $provider->getName(),
+            return $this->redirectToInitiationForm(
+                $provider,
                 ['authenticationFailed' => true]
```

Comment:
Why pass the provider alongside the options when all you do in the `redirectToInitiationForm` is: adding the provider name to the options?

Reply:
We could pass a string, but then I can't be sure the string refers to a valid GSSP provider.